api/signup_test.go

```diff
@@ -0,0 +1,141 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/netlify/gotrue/models"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type SignupTestSuite struct {
+	suite.Suite
+	API *API
+}
+
+func (ts *SignupTestSuite) SetupTest() {
+	api, err := NewAPIFromConfigFile("config.test.json", "v1")
+	require.NoError(ts.T(), err)
+
+	ts.API = api
+
+	// Cleanup existing user
+	u, err := ts.API.db.FindUserByEmailAndAudience("test@example.com", api.config.JWT.Aud)
+	if err == nil {
+		require.NoError(ts.T(), api.db.DeleteUser(u))
+	}
+}
+
+// TestSignup tests API /signup route
+func (ts *SignupTestSuite) TestSignup() {
+	// Request body
+	var buffer bytes.Buffer
+	require.NoError(ts.T(), json.NewEncoder(&buffer).Encode(map[string]interface{}{
+		"email":    "test@example.com",
+		"password": "test",
+		"data": map[string]interface{}{
+			"a": 1,
+		},
+	}))
+
+	// Setup request
+	req := httptest.NewRequest("POST", "http://localhost/signup", &buffer)
+	req.Header.Set("Content-Type", "application/json")
+
+	// Setup response recorder
+	w := httptest.NewRecorder()
+	ctx := req.Context()
+
+	ts.API.Signup(ctx, w, req)
+
+	resp := w.Result()
+	assert.Equal(ts.T(), resp.StatusCode, 200)
+
+	data := make(map[string]interface{})
+	require.NoError(ts.T(), json.NewDecoder(resp.Body).Decode(&data))
+	assert.Equal(ts.T(), data["email"], "test@example.com")
```

Comment:
I would also check the size of the map returned. So that you know what you expect to come back.